extensions/vscode-api-tests/src/workspace.test.ts

```diff
@@ -160,6 +160,41 @@ suite('workspace-namespace', () => {
 		});
 	});
 
+	test('eol, read', () => {
+		const a = createRandomFile('foo\nbar\nbar').then(file => {
+			return workspace.openTextDocument(file).then(doc => {
+				assert.equal(doc.eol, EndOfLine.LF);
+			});
+		});
+		const b = createRandomFile('foo\nbar\nbar\r\nbaz').then(file => {
+			return workspace.openTextDocument(file).then(doc => {
+				assert.equal(doc.eol, EndOfLine.LF);
+			});
+		});
+		const c = createRandomFile('foo\r\nbar\r\nbar').then(file => {
+			return workspace.openTextDocument(file).then(doc => {
+				assert.equal(doc.eol, EndOfLine.CRLF);
+			});
+		});
+		return Promise.all([a, b, c]);
+	});
+
+	test('eol, change via editor', () => {
+		return createRandomFile('foo\nbar\nbar').then(file => {
+			return workspace.openTextDocument(file).then(doc => {
+				assert.equal(doc.eol, EndOfLine.LF);
+				return window.showTextDocument(doc).then(editor => {
+					return editor.edit(builder => builder.setEndOfLine(EndOfLine.CRLF));
```

Comment:
You are not looking the new API. The edit-builder allows since a while to set the eol-sequence, but only for an editor, not generically for a document. That is yet to be defined.